rs/replicated_state/src/canister_state/system_state/call_context_manager.rs

```diff
@@ -917,6 +929,36 @@ impl CallContextManager {
         );
         true
     }
+
+    /// Marks all call contexts as deleted and produces reject responses for the
+    /// not yet responded ones. This is called as part of uninstalling a canister.
+    ///
+    /// Callbacks will be unregistered when responses are received.
+    pub(super) fn delete_all_call_contexts<R>(
+        &mut self,
+        reject: impl Fn(&CallContext) -> Option<R>,
+    ) -> Vec<R> {
+        let mut reject_responses = Vec::new();
+
+        for call_context in self.call_contexts.values_mut() {
+            // Mark the call context as deleted.
+            call_context.mark_deleted();
+
+            if !call_context.has_responded() {
+                // Generate a reject response.
+                if let Some(response) = reject(call_context) {
+                    reject_responses.push(response)
+                }
+
+                call_context.mark_responded();
```

Comment:
Compared to the old code we are marking the call context as deleted here as opposed to a separate block below. Is this ok?

Reply:
Technically a call context must have a response before it can be deleted (which is one of the things that this method ensures). So marking it deleted first and responded afterwards is not ideal. It has no material effect on anything, but it does unnecessarily raise the question of why.

So I swapped it around to how it was.